src/ImageSharp/Processing/Processors/Quantization/DefaultPixelSamplingStrategy.cs

```diff
@@ -101,4 +101,54 @@ Buffer2DRegion<TPixel> GetRow(int pos)
             }
         }
     }
+
+    /// <inheritdoc />
+    public IEnumerable<Buffer2DRegion<TPixel>> EnumeratePixelRegions<TPixel>(ImageFrame<TPixel> frame)
+        where TPixel : unmanaged, IPixel<TPixel>
+    {
+        long maximumPixels = Math.Min(this.MaximumPixels, (long)frame.Width * frame.Height);
+        long maxNumberOfRows = maximumPixels / frame.Width;
+        long totalNumberOfRows = frame.Height;
+
+        if (totalNumberOfRows <= maxNumberOfRows)
+        {
+            yield return frame.PixelBuffer.GetRegion();
+        }
+        else
+        {
+            double r = maxNumberOfRows / (double)totalNumberOfRows;
+
+            // Use a rough approximation to make sure we don't leave out large contiguous regions:
+            if (maxNumberOfRows > 200)
+            {
+                r = Math.Round(r, 2);
+            }
+            else
+            {
+                r = Math.Round(r, 1);
+            }
+
+            r = Math.Max(this.MinimumScanRatio, r); // always visit the minimum defined portion of the image.
+
+            Rational ratio = new(r);
+
+            int denom = (int)ratio.Denominator;
+            int num = (int)ratio.Numerator;
+
+            for (int pos = 0; pos < totalNumberOfRows; pos++)
+            {
+                int subPos = pos % denom;
```

Comment:
```suggestion
                Debug.Assert(denom > 0);
                int subPos = (int)((uint)pos % (uint)denom);
```
Emits a "better" divison as the sign doesn't need to be considered.
Same below.

Although in contrast of an IEnumerable that effect won't be big.
If this method is hot, maybe a custom struct-based enumerator has more impact on perf and allocations.